miden-tx/src/testing/tx_context.rs

```diff
@@ -197,21 +216,385 @@ impl TransactionContextBuilder {
+            aux0 = output0.metadata().aux(),
+            tag0 = output0.metadata().tag(),
+            asset0 = prepare_assets(output0.assets())[0],
+            recipient1 = prepare_word(&output1.recipient().digest()),
+            aux1 = output1.metadata().aux(),
+            tag1 = output1.metadata().tag(),
+            asset1 = prepare_assets(output1.assets())[0],
+        );
+
+        NoteBuilder::new(sender, ChaCha20Rng::from_seed(self.rng.gen()))
+            .note_inputs(inputs)
+            .unwrap()
+            .add_assets([asset])
+            .code(code)
+            .build(&self.assembler)
+            .unwrap()
+    }
+
+    fn input_note_transfer(
+        &mut self,
+        sender: AccountId,
+        assets: impl IntoIterator<Item = Asset>,
+    ) -> Note {
+        let code = "
+            use.miden::note
+            use.miden::contracts::wallets::basic->wallet
+
+            begin
+                # read the assets to memory
+                push.0 exec.note::get_assets
+                # => [num_assets, dest_ptr]
+
+                # assert the number of assets is 3
+                push.3 assert_eq
+                # => [dest_ptr]
+
+                # add the first asset to the vault
+                padw dup.4 mem_loadw call.wallet::receive_asset dropw
+                # => [dest_ptr]
+
+                # add the second asset to the vault
+                push.1 add padw dup.4 mem_loadw call.wallet::receive_asset dropw
+                # => [dest_ptr+1]
+
+                # add the third asset to the vault
+                push.1 add padw movup.4 mem_loadw call.wallet::receive_asset dropw
+                # => []
+            end
+        ";
+
+        NoteBuilder::new(sender, ChaCha20Rng::from_seed(self.rng.gen()))
+            .add_assets(assets)
+            .code(code)
+            .build(&self.assembler)
+            .unwrap()
+    }
+
+    pub fn with_mock_notes_too_few_input(mut self) -> Self {
+        // ACCOUNT IDS
+        // --------------------------------------------------------------------------------------------
+        let sender = AccountId::try_from(ACCOUNT_ID_SENDER).unwrap();
+        let faucet_id_1 = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_1).unwrap();
+        let faucet_id_2 = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_2).unwrap();
+        let faucet_id_3 = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_3).unwrap();
+
+        // ASSETS
+        // --------------------------------------------------------------------------------------------
+        let fungible_asset_1: Asset =
+            FungibleAsset::new(faucet_id_1, CONSUMED_ASSET_1_AMOUNT).unwrap().into();
+        let fungible_asset_2: Asset =
+            FungibleAsset::new(faucet_id_2, CONSUMED_ASSET_2_AMOUNT).unwrap().into();
+        let fungible_asset_3: Asset =
+            FungibleAsset::new(faucet_id_3, CONSUMED_ASSET_3_AMOUNT).unwrap().into();
```

Comment:
These are the constants I was trying to define using `const` functions. Unfortunately because of the validation/unwrap it is not possible yet.